docs/docs/concepts/05-building-blocks/01-signals.md

```diff
@@ -0,0 +1,88 @@
+---
+sidebar_position: 1
+---
+
+# Signals
+
+In the AICA framework, signals are designed to exchange continuous data between components and controllers in a running
+application. They are an abstraction of ROS 2 topics.
+
+ROS 2 topics are messages sent from publishers to subscribers under a specific namespace (the message "topic")
+using some predefined message format (the message "type"). As a result, topics enable ROS nodes to communicate in many
+different network topologies and message formats.
+
+Often, the message topic and message type is hardcoded within the implementation of a ROS node. This can make it
+difficult to rearrange the network topology of ROS nodes without modifying and recompiling the node implementation
+itself.
+
+Signals then are ROS 2 publishers and subscribers with dynamically assigned topics and standardized message types.
+This makes it easy to reconfigure the signal connections between different components and controllers in the application
+graph without modifying or recompiling any source code. By using standard message types, the signal compatibility
+between components and controllers is also simplified.
+
+<!-- TODO: not sure if this is too much to put here, since it's more relevant to the programming reference
```

Comment:
Hmm I do think that this could be relevant, maybe in combination with a rewrite of the State messages section below (see comment).